src/server/cluster/cluster_family.cc

```diff
@@ -589,6 +593,72 @@ void ClusterFamily::DflyClusterFlushSlots(CmdArgList args, ConnectionContext* cn
   return rb->SendOk();
 }
 
+void ClusterFamily::DflyClusterStartSlotMigration(CmdArgList args, ConnectionContext* cntx) {
+  SinkReplyBuilder* rb = cntx->reply_builder();
+
+  args.remove_prefix(1);  // Removes "START-SLOT-MIGRATION" subcmd string
+
+  CmdArgParser parser(args);
+  auto [host_ip, port] = parser.Next<std::string_view, uint16_t>();
+  std::vector<SlotRange> slots;
+  do {
+    auto [slot_start, slot_end] = parser.Next<SlotId, SlotId>();
+    slots.emplace_back(SlotRange{slot_start, slot_end});
+  } while (parser.HasNext());
+
+  if (auto err = parser.Error(); err)
+    return (*cntx)->SendError(err->MakeReply());
+
+  ClusterSlotMigration node(std::string(host_ip), port, slots);
+  node.Start(cntx);
+
+  return rb->SendOk();
+}
+
+void ClusterFamily::DflyMigrate(CmdArgList args, ConnectionContext* cntx) {
+  ToUpper(&args[0]);
+  string_view sub_cmd = ArgS(args, 0);
+  args.remove_prefix(1);
+  if (sub_cmd == "CONF") {
+    MigrationConf(args, cntx);
+  } else {
+    (*cntx)->SendError(facade::UnknownSubCmd(sub_cmd, "DFLYMIGRATE"), facade::kSyntaxErrType);
+  }
+}
+
+void ClusterFamily::MigrationConf(CmdArgList args, ConnectionContext* cntx) {
+  VLOG(1) << "Create slot migration config";
+  CmdArgParser parser{args};
+  auto port = parser.Next<uint16_t>();
+  (void)port;  // we need it for the next step
+
+  std::vector<ClusterConfig::SlotRange> slots;
+  do {
+    auto [slot_start, slot_end] = parser.Next<SlotId, SlotId>();
+    slots.emplace_back(SlotRange{slot_start, slot_end});
+  } while (parser.HasNext());
+
+  if (!tl_cluster_config) {
+    (*cntx)->SendError(kClusterNotConfigured);
+    return;
+  }
+
+  for (const auto& migration_range : slots) {
+    for (auto i = migration_range.start; i <= migration_range.end; ++i) {
+      if (!tl_cluster_config->IsMySlot(i)) {
+        VLOG(1) << "Invalid migration slot rane";
```

Comment:
nit: rane -> range
also print the invalid slot id, the one that is not in the range